crypto/asn1/tasn_new.c

```diff
@@ -147,6 +147,8 @@ int asn1_item_embed_new(ASN1_VALUE **pval, const ASN1_ITEM *it, int embed)
     return 1;
 
  memerr2:
+    if (asn1_cb != NULL)
+        (void)asn1_cb(ASN1_OP_NEW_POST, pval, it, NULL);
```

Comment:
I am not convinced this is the correct fix. It changes the semantics of `ASN1_OP_NEW_POST` which now has to be tolerant of a partially constructed object. We can confirm our own code does this but this could be viewed as a breaking change for external users of this. On the other hand I think it is perfect reasonable for `ASN1_OP_FREE_POST` to have to be tolerant of partially constructed objects. This seems normal behaviour for a "free" function. Indeed this is the case now in the code so this isn't a change in semantics for this operation. However it just so happens that `ASN1_OP_FREE_POST` for CRLs is not tolerant of this - and that is the cause of the bug.

I think instead of this fix, the ASN1_OP_FREE_POST in x_crl.c should be changed to check whether `crl->meth` is NULL or not, and not call the crl_free function if not. This should only be NULL in the case where we have failed during initial construction, and the default CRL_METHOD does nothing on crl_free anyway - so this should be a safe fix.

Reply:
Hmm, you are probably right, the only other place that looks suspicious is this:
crypto/x509/x_x509.c#L59-L95
```C
    case ASN1_OP_NEW_POST:
        ret->ex_cached = 0;
        [...]
        ret->aux = NULL;
        ret->crldp = NULL;
        if (!CRYPTO_new_ex_data(CRYPTO_EX_INDEX_X509, ret, &ret->ex_data))
            return 0;
        break;


    case ASN1_OP_FREE_POST:
        CRYPTO_free_ex_data(CRYPTO_EX_INDEX_X509, ret, &ret->ex_data);
        X509_CERT_AUX_free(ret->aux);
        ASN1_OCTET_STRING_free(ret->skid);
        [...]
        break;
```
where the `ret->ex_data` is not initialized with `CRYPTO_new_ex_data` but just zeroed,
and that is probably safe.